CloudStorageCore/src/main/java/com/gkatzioura/maven/cloud/resolver/KeyResolver.java

```diff
@@ -18,25 +18,25 @@
 
 public class KeyResolver {
 
-    public String resolve(String... paths) {
-
-        StringBuilder builder = new StringBuilder();
+    public String resolve(String base, String path) {
```

Comment:
This change happened because I noticed that I needed to preserve trailing slashes for directory-or-not checking, as the way wagon-maven-plugin checks if something is a directory is by checking if it, with a trailing slash, exists. This seems correct per the Wagon contract. Since resolve was stripping the trailing slash, files were returning as existing, which was interpreted as a directory existing, and wagon-maven-plugin tried to recurse into them.

Initially I started to modify the method minimally, but nothing I was writing seemed clean. I noticed that all non-test uses of this method use two arguments, so I modified it to be just a two argument method and wrote an implementation for that.